### DISCUSSION TITLE
No such module 'ComposableArchitecture'

### DISCUSSION THREAD

**Discussion by danlee2025** [pointfreeco/swift-composable-architecture#948] [Q&A] [answered]
Hello, thank you for your amazing library, recently I met this problem.
I am using Xcode 13.1, here's the way to reproduce the problem:
First, create a "multiplatform app"
Second, change target to macos
Third, add composable library in SPM
Fourth,in the code import this library and then compile the project, then it will result in an error complaining about "No such module"
but the ios target building is ok. Why was that?

**Comment by AlexisQapa** [comment, marked as answer]
If you add the lib via the Xcode packages interface, Xcode asks you which target you want to add the lib to (but only one). Juste go to your target settings and add a dependency to the lib.

**Comment by stephencelis** [comment]
Hi @danleechina! I'm going to convert this to a discussion, since it doesn't appear to be a bug with the library. For a visual, @AlexisQapa means you must add the library to the Mac target here (under **Frameworks, Libraries, and Embedded Content**):

<img width="584" alt="image" src="https://user-images.githubusercontent.com/658/148230679-d4db7bf1-1eff-4f5a-9036-dc715dc667fa.png">

**Reply by macinsmith** [reply]
This continues to be a problem.  I created a fresh SwiftUI project, went to the App target and added the SP for ComposableArchitecture.  I let it default to the latest which is 1.9.2.  I then added it to my test targets as well.  When I import it I still get the message "No such module 'ComposableArchitecture'.  I'm sure the package is fine and this is Apple's bug, but does anyone know how to get around it?  I'm running Xcode 15.2 under Sonoma 14.4 and my project is an iOS app

![Screenshot 2024-03-21 at 12 19 50 PM](https://github.com/pointfreeco/swift-composable-architecture/assets/3606154/1b749043-4308-4894-b1e6-f042c285286c)
![Screenshot 2024-03-21 at 12 23 35 PM](https://github.com/pointfreeco/swift-composable-architecture/assets/3606154/918e274c-8439-42b0-bd43-4f61e86fa7f2)

**Reply by mbrandonw** [reply]
Hi @macinsmith, are you sure you have enabled macros before building? I would click the red X icon in the top-right of the Xcode window to make sure. 

And if you have definitely done that, then can you provide a minimal project that demonstrates the problem? That makes it easier to look into the issue.

**Reply by macinsmith** [reply]
That was the issue.   Fixed!

**Comment by macinsmith** [comment]
That was it, however it didn’t fix it immediately.  It took a few seconds for it to show up. 

Sorry to have bothered you.  I will add that to my comment on the issue in the repo.

Steve

On Mar 21, 2024, at 7:51 PM, Brandon Williams ***@***.***> wrote:


Hi @macinsmith <https://github.com/macinsmith>, are you sure you have enabled macros before building? I would click the red X icon in the top-right of the Xcode window to make sure.

And if you have definitely done that, then can you provide a minimal project that demonstrates the problem? That makes it easier to look into the issue.
